servicetalk-concurrent-api-internal/src/main/java/io/servicetalk/concurrent/api/internal/ConnectablePayloadWriter.java

```diff
@@ -329,7 +327,11 @@ public void request(final long n) {
 
                     @Override
                     public void cancel() {
-                        if (closedUpdater.compareAndSet(outer, null, error(CONNECTED_PUBLISHER_CANCELLED))) {
+                        if (outer.closed != null) {
```

Comment:
is that changing the semantics compared to not checking previously?

Reply:
I don't think so. The code below is doing something only if `outer.closed == null`. Otherwise, it's noop. Because previously CAS attempt did not allocate anything, this check was not necessary. Now it helps to avoid unnecessary allocation if the state is not null anymore -> we know CAS will fail.